File: examples/NSTouchBarCatalogCreatingandCustomizingNSTouchBar/Objective-C/NSTouchBar Catalog/WindowController.m
/*
 Copyright (C) 2016 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 Main window controller for this sample.
 */

#import "WindowController.h"

static NSTouchBarItemIdentifier WindowControllerLabelIdentifier = @"com.TouchBarCatalog.windowController.label";

@interface WindowController () <NSTouchBarDelegate>

@end


#pragma mark -

@implementation WindowController

- (void)windowDidLoad
{
    [super windowDidLoad];
    
    self.window.frameAutosaveName = @"WindowAutosave";
}

// This window controller will have only one NSTouchBarItem instance, which is a simple label,
// so to show that view controller bar can reside along side its window controller.
//
- (NSTouchBar *)makeTouchBar
{
    NSTouchBar *bar = [[NSTouchBar alloc] init];
    bar.delegate = self;
        
    // Set the default ordering of items.
    bar.defaultItemIdentifiers =
        @[WindowControllerLabelIdentifier, NSTouchBarItemIdentifierOtherItemsProxy];
    
    return bar;
}

- (nullable NSTouchBarItem *)touchBar:(NSTouchBar *)touchBar makeItemForIdentifier:(NSTouchBarItemIdentifier)identifier
{
    if ([identifier isEqualToString:WindowControllerLabelIdentifier])
    {
        NSTextField *theLabel = [NSTextField labelWithString:@"Catalog"];
        
        NSCustomTouchBarItem *customItemForLabel =
            [[NSCustomTouchBarItem alloc] initWithIdentifier:WindowControllerLabelIdentifier];
        customItemForLabel.view = theLabel;
        
        // We want this label to always be visible no matter how many items are in the NSTouchBar instance.
        customItemForLabel.visibilityPriority = NSTouchBarItemPriorityHigh;
        
        return customItemForLabel;
    }
    
    return nil;
}

@end

File: examples/NSTouchBarCatalogCreatingandCustomizingNSTouchBar/Swift/NSTouchBar Catalog/WindowController.swift
/*
 Copyright (C) 2016 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 Main window controller for this sample.
 */

import Cocoa

enum AssetNames: String {
    case accounts = "Accounts", bookmark = "Bookmark", cat = "DorianGrayCat",
         red = "Red", green = "Green"
}

fileprivate extension NSTouchBarCustomizationIdentifier {
    static let windowBar = NSTouchBarCustomizationIdentifier("com.TouchBarCatalog.windowTouchBar")
}

fileprivate extension NSTouchBarItemIdentifier {
    static let label = NSTouchBarItemIdentifier("com.TouchBarCatalog.TouchBarItem.label")
}

class WindowController: NSWindowController {

    override func windowDidLoad() {
        super.windowDidLoad()
        
        self.window?.setFrameAutosaveName("WindowAutosave")
    }
    
    // MARK: NSTouchBar
    
    override func makeTouchBar() -> NSTouchBar? {
        let touchBar = NSTouchBar()
        touchBar.delegate = self
        touchBar.customizationIdentifier = .windowBar
        touchBar.defaultItemIdentifiers = [.label, .fixedSpaceLarge, .otherItemsProxy]
        touchBar.customizationAllowedItemIdentifiers = [.label]
        
        return touchBar
    }
    
}

// MARK: NSTouchBarDelegate

extension WindowController: NSTouchBarDelegate {
    
    func touchBar(_ touchBar: NSTouchBar, makeItemForIdentifier identifier: NSTouchBarItemIdentifier) -> NSTouchBarItem? {
        
        switch identifier {
        case NSTouchBarItemIdentifier.label:
            let custom = NSCustomTouchBarItem(identifier: identifier)
            custom.customizationLabel = "TouchBar Catalog Label"
            
            let label = NSTextField.init(labelWithString: "Catalog")
            custom.view = label
            
            return custom
            
        default:
            return nil
        }
    }
}
